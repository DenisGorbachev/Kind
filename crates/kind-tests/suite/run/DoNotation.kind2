Main : Data.Maybe Data.U60
Main =
    do Data.Maybe {
        Data.Maybe.some 3
        Data.Maybe.pure 2
        ask res  = Data.Maybe.pure 2
        ask res2 = Data.Maybe.pure 3
        match Data.Maybe t = (Data.Maybe.some 4) {
            some val => Data.Maybe.pure (+ 1000 (+ val (+ res res2)))
            none     => Data.Maybe.none
        }
    }