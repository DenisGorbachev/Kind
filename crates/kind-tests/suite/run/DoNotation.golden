(Data.Maybe.some 1009)